File: binding/include/libqtextboundaryfinder.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qtextboundaryfinder.html
pub const qtextboundaryfinder = struct {

    /// New constructs a new QTextBoundaryFinder object.
    pub fn New() QtC.QTextBoundaryFinder {
        return qtc.QTextBoundaryFinder_new();
    }


    /// New2 constructs a new QTextBoundaryFinder object.
    pub fn New2(other: ?*anyopaque) QtC.QTextBoundaryFinder {
        return qtc.QTextBoundaryFinder_new2(@ptrCast(other));
    }


    /// New3 constructs a new QTextBoundaryFinder object.
    pub fn New3(typeVal: i32, stringVal: []const u8) QtC.QTextBoundaryFinder {
        const stringVal_str = qtc.libqt_string{
    .len = stringVal.len,
    .data = stringVal.ptr,
};
return qtc.QTextBoundaryFinder_new3(@intCast(typeVal), stringVal_str);
    }


    /// New4 constructs a new QTextBoundaryFinder object.
    pub fn New4(typeVal: i32, chars: ?*anyopaque, length: i64) QtC.QTextBoundaryFinder {
        return qtc.QTextBoundaryFinder_new4(@intCast(typeVal), @ptrCast(chars), length);
    }


    /// New5 constructs a new QTextBoundaryFinder object.
    pub fn New5(typeVal: i32, str: []const u8) QtC.QTextBoundaryFinder {
        return qtc.QTextBoundaryFinder_new5(@intCast(typeVal), @ptrCast(str));
    }


    /// New6 constructs a new QTextBoundaryFinder object.
    pub fn New6(typeVal: i32, chars: ?*anyopaque, length: i64, buffer: *u8) QtC.QTextBoundaryFinder {
        return qtc.QTextBoundaryFinder_new6(@intCast(typeVal), @ptrCast(chars), length, @ptrCast(buffer));
    }


    /// New7 constructs a new QTextBoundaryFinder object.
    pub fn New7(typeVal: i32, chars: ?*anyopaque, length: i64, buffer: *u8, bufferSize: i64) QtC.QTextBoundaryFinder {
        return qtc.QTextBoundaryFinder_new7(@intCast(typeVal), @ptrCast(chars), length, @ptrCast(buffer), bufferSize);
    }


    /// New8 constructs a new QTextBoundaryFinder object.
    pub fn New8(typeVal: i32, str: []const u8, buffer: *u8) QtC.QTextBoundaryFinder {
        return qtc.QTextBoundaryFinder_new8(@intCast(typeVal), @ptrCast(str), @ptrCast(buffer));
    }


    /// New9 constructs a new QTextBoundaryFinder object.
    pub fn New9(typeVal: i32, str: []const u8, buffer: *u8, bufferSize: i64) QtC.QTextBoundaryFinder {
        return qtc.QTextBoundaryFinder_new9(@intCast(typeVal), @ptrCast(str), @ptrCast(buffer), bufferSize);
    }


    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QTextBoundaryFinder_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    pub fn IsValid(self: ?*anyopaque, ) bool {
        return qtc.QTextBoundaryFinder_IsValid(@ptrCast(self));
    }

    pub fn Type(self: ?*anyopaque, ) i32 {
        return qtc.QTextBoundaryFinder_Type(@ptrCast(self));
    }

    pub fn String(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QTextBoundaryFinder_String(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qtextboundaryfinder.String: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn ToStart(self: ?*anyopaque, ) void {
        qtc.QTextBoundaryFinder_ToStart(@ptrCast(self));
    }

    pub fn ToEnd(self: ?*anyopaque, ) void {
        qtc.QTextBoundaryFinder_ToEnd(@ptrCast(self));
    }

    pub fn Position(self: ?*anyopaque, ) i64 {
        return qtc.QTextBoundaryFinder_Position(@ptrCast(self));
    }

    pub fn SetPosition(self: ?*anyopaque, position: i64) void {
        qtc.QTextBoundaryFinder_SetPosition(@ptrCast(self), position);
    }

    pub fn ToNextBoundary(self: ?*anyopaque, ) i64 {
        return qtc.QTextBoundaryFinder_ToNextBoundary(@ptrCast(self));
    }

    pub fn ToPreviousBoundary(self: ?*anyopaque, ) i64 {
        return qtc.QTextBoundaryFinder_ToPreviousBoundary(@ptrCast(self));
    }

    pub fn IsAtBoundary(self: ?*anyopaque, ) bool {
        return qtc.QTextBoundaryFinder_IsAtBoundary(@ptrCast(self));
    }

    pub fn BoundaryReasons(self: ?*anyopaque, ) i32 {
        return qtc.QTextBoundaryFinder_BoundaryReasons(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QTextBoundaryFinder_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/textboundaryfinder.html#types
pub const enums = struct {
    pub const BoundaryType = enum {
        pub const Grapheme: i32 = 0;
        pub const Word: i32 = 1;
        pub const Sentence: i32 = 2;
        pub const Line: i32 = 3;
    };

    pub const BoundaryReason = enum {
        pub const NotAtBoundary: i32 = 0;
        pub const BreakOpportunity: i32 = 31;
        pub const StartOfItem: i32 = 32;
        pub const EndOfItem: i32 = 64;
        pub const MandatoryBreak: i32 = 128;
        pub const SoftHyphen: i32 = 256;
    };

};

File: binding/src/libqtextboundaryfinder.cpp
#include <QChar>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QStringView>
#include <QTextBoundaryFinder>
#include <qtextboundaryfinder.h>
#include "libqtextboundaryfinder.h"
#include "libqtextboundaryfinder.hxx"

QTextBoundaryFinder* QTextBoundaryFinder_new() {
	 return new QTextBoundaryFinder();
}

QTextBoundaryFinder* QTextBoundaryFinder_new2(const QTextBoundaryFinder* other) {
	 return new QTextBoundaryFinder(*other);
}

QTextBoundaryFinder* QTextBoundaryFinder_new3(int typeVal, const libqt_string stringVal) {
	 return new QTextBoundaryFinder(static_cast<QTextBoundaryFinder::BoundaryType>(typeVal), QString::fromUtf8(stringVal.data, stringVal.len));
}

QTextBoundaryFinder* QTextBoundaryFinder_new4(int typeVal, const QChar* chars, ptrdiff_t length) {
	 return new QTextBoundaryFinder(static_cast<QTextBoundaryFinder::BoundaryType>(typeVal), chars, length);
}

QTextBoundaryFinder* QTextBoundaryFinder_new5(int typeVal, QStringView* str) {
	 return new QTextBoundaryFinder(static_cast<QTextBoundaryFinder::BoundaryType>(typeVal), *str);
}

QTextBoundaryFinder* QTextBoundaryFinder_new6(int typeVal, const QChar* chars, ptrdiff_t length, unsigned char* buffer) {
	 return new QTextBoundaryFinder(static_cast<QTextBoundaryFinder::BoundaryType>(typeVal), chars, length, buffer);
}

QTextBoundaryFinder* QTextBoundaryFinder_new7(int typeVal, const QChar* chars, ptrdiff_t length, unsigned char* buffer, ptrdiff_t bufferSize) {
	 return new QTextBoundaryFinder(static_cast<QTextBoundaryFinder::BoundaryType>(typeVal), chars, length, buffer, bufferSize);
}

QTextBoundaryFinder* QTextBoundaryFinder_new8(int typeVal, QStringView* str, unsigned char* buffer) {
	 return new QTextBoundaryFinder(static_cast<QTextBoundaryFinder::BoundaryType>(typeVal), *str, buffer);
}

QTextBoundaryFinder* QTextBoundaryFinder_new9(int typeVal, QStringView* str, unsigned char* buffer, ptrdiff_t bufferSize) {
	 return new QTextBoundaryFinder(static_cast<QTextBoundaryFinder::BoundaryType>(typeVal), *str, buffer, bufferSize);
}

void QTextBoundaryFinder_OperatorAssign(QTextBoundaryFinder* self, const QTextBoundaryFinder* other) {
	self->operator=(*other);
}

bool QTextBoundaryFinder_IsValid(const QTextBoundaryFinder* self) {
	return self->isValid();
}

int QTextBoundaryFinder_Type(const QTextBoundaryFinder* self) {
	return self->type();
}

libqt_string QTextBoundaryFinder_String(const QTextBoundaryFinder* self) {
	QString _ret = self->string();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QTextBoundaryFinder_ToStart(QTextBoundaryFinder* self) {
	self->toStart();
}

void QTextBoundaryFinder_ToEnd(QTextBoundaryFinder* self) {
	self->toEnd();
}

ptrdiff_t QTextBoundaryFinder_Position(const QTextBoundaryFinder* self) {
	return self->position();
}

void QTextBoundaryFinder_SetPosition(QTextBoundaryFinder* self, ptrdiff_t position) {
	self->setPosition(position);
}

ptrdiff_t QTextBoundaryFinder_ToNextBoundary(QTextBoundaryFinder* self) {
	return self->toNextBoundary();
}

ptrdiff_t QTextBoundaryFinder_ToPreviousBoundary(QTextBoundaryFinder* self) {
	return self->toPreviousBoundary();
}

bool QTextBoundaryFinder_IsAtBoundary(const QTextBoundaryFinder* self) {
	return self->isAtBoundary();
}

int QTextBoundaryFinder_BoundaryReasons(const QTextBoundaryFinder* self) {
	return self->boundaryReasons();
}

void QTextBoundaryFinder_Delete(QTextBoundaryFinder* self) {
    delete self;
}

